### PR TITLE
[ci-visibility] Add early flake detection to cucumber 

### PR DESCRIPTION
**Pull request by juan-fernandez**
### What does this PR do?
Following https://github.com/DataDog/dd-trace-js/pull/3956 and https://github.com/DataDog/dd-trace-js/pull/4060, this PR implements early flake detection logic for cucumber.

### Motivation
Allow customers to detect their flaky tests before they're in the default branch.

### Plugin Checklist

- [x] Unit tests.

### Security 
Datadog employees:
- [ ] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

Unsure? Have a question? Request a review!

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -27,7 +27,7 @@
 /integration-tests/cucumber/ @DataDog/ci-app-libraries
 /integration-tests/cypress/ @DataDog/ci-app-libraries
 /integration-tests/playwright/ @DataDog/ci-app-libraries
-/integration-tests/ci-visibility-spec.js @DataDog/ci-app-libraries
+/integration-tests/ci-visibility.spec.js @DataDog/ci-app-libraries
```
Comment (juan-fernandez) [inline CODEOWNERS:30]: unrelated fix

File: integration-tests/ci-visibility.spec.js [hunk 2]
```diff
@@ -744,7 +744,7 @@ testFrameworks.forEach(({
 
               const tests = events.filter(event => event.type === 'test').map(event => event.content)
               const newTests = tests.filter(test =>
-                test.meta[TEST_SUITE] === 'true'
+                test.meta[TEST_IS_NEW] === 'true'
```
Comment (juan-fernandez) [inline integration-tests/ci-visibility.spec.js:747]: unrelated fix

**Review by rochdev — APPROVED** [review]

**Review by anmarchenko — APPROVED** [review]
